app/src/main/java/io/appium/settings/handlers/BluetoothConnectionSettingHandler.java

```diff
@@ -0,0 +1,38 @@
+/*
+  Copyright 2012-present Appium Committers
+  <p>
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  <p>
+  http://www.apache.org/licenses/LICENSE-2.0
+  <p>
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+package io.appium.settings.handlers;
+
+import android.bluetooth.BluetoothAdapter;
+import android.content.Context;
+
+public class BluetoothConnectionSettingHandler extends AbstractSettingHandler {
+
+    public BluetoothConnectionSettingHandler(Context context) {
+        super(context);
+    }
+
+    @Override
+    protected boolean setState(boolean state) {
+        BluetoothAdapter bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
```

Comment:
Can bluetoothAdapter  be `null`?

Reply:
good point

> BluetoothAdapter | the default local adapter, or null if Bluetooth is not supported on this hardware  platform

Reply:
added null check in AbstractSettingReceiver